asciidoctorj-core/src/main/java/org/asciidoctor/ast/AbstractNodeImpl.java

```diff
@@ -54,157 +63,211 @@ public DocumentRuby document() {
 
     @Override
     public DocumentRuby getDocument() {
-        return this.abstractNode.getDocument();
+        return (DocumentRuby) NodeConverter.createASTNode(getRubyProperty("document"));
+    }
+
+    public IRubyObject getRubyObject() {
+        return rubyNode;
```

Comment:
I'm at the phone so I cannot look at the code right now. 
But the method should only be available in the Impl class, not in the interface. 
So the user of the API will not see this method. 



> Am 05.06.2015 um 23:22 schrieb Dan Allen <notifications@github.com>:
> 
> In asciidoctorj-core/src/main/java/org/asciidoctor/ast/AbstractNodeImpl.java:
> 
> > @@ -54,157 +63,211 @@ public DocumentRuby document() {
> >  
> >      @Override
> >      public DocumentRuby getDocument() {
> > -        return this.abstractNode.getDocument();
> > +        return (DocumentRuby) NodeConverter.createASTNode(getRubyProperty("document"));
> > +    }
> > +
> > +    public IRubyObject getRubyObject() {
> > +        return rubyNode;
> I think getRubyObject should be either protected or package-private. I'm not sure we should have it exposed (yet) as a public API method.
> 
> —
> Reply to this email directly or view it on GitHub.
>